internal/engine/dockerprune/docker_pruner.go

```diff
@@ -149,13 +156,52 @@ func (dp *DockerPruner) prune(ctx context.Context, maxAge time.Duration) error {
 		prettyPrintCachePruneReport(cacheReport, l)
 	}
 
-	containerReport, err := dp.dCli.ContainersPrune(ctx, f)
+	return nil
+}
+
+func (dp *DockerPruner) deleteOldImages(ctx context.Context, maxAge time.Duration) (types.ImagesPruneReport, error) {
+	opts := types.ImageListOptions{
+		Filters: filters.NewArgs(
+			filters.Arg("label", docker.BuiltByTiltLabelStr),
+		),
+	}
+	imgs, err := dp.dCli.ImageList(ctx, opts)
 	if err != nil {
-		return err
+		return types.ImagesPruneReport{}, err
 	}
-	prettyPrintContainersPruneReport(containerReport, l)
 
-	return nil
+	toDelete := make(map[string]uint64) // map imageID to size in bytes
+	for _, imgSummary := range imgs {
+		inspect, _, err := dp.dCli.ImageInspectWithRaw(ctx, imgSummary.ID)
+		if err != nil {
+			logger.Get(ctx).Debugf("[Docker prune] error inspecting image '%s': %v", imgSummary.ID, err)
+			continue
+		}
+
+		if time.Since(inspect.Metadata.LastTagTime) >= maxAge {
+			toDelete[inspect.ID] = uint64(inspect.Size)
+		}
+	}
+
+	rmOpts := types.ImageRemoveOptions{PruneChildren: true}
+	var responseItems []types.ImageDeleteResponseItem
+	var reclaimedBytes uint64
+
+	for imgID, bytes := range toDelete {
+		items, err := dp.dCli.ImageRemove(ctx, imgID, rmOpts)
+		// No good way to detect in-use images from `inspect` output, so just silently ignore
+		if err != nil && !strings.Contains(err.Error(), "image is being used by running container") {
```

Comment:
if the image is being used by a running container, that means we shouldn't report it as deleted or reclaimed, right?

Reply:
whoops, right!